### PR TITLE
Feature findings enhancemnt

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
Findings API Enhancements
 
### Issues Resolved
[[Issue](https://github.com/opensearch-project/security-analytics/issues/795)
Alerting PR: https://github.com/opensearch-project/alerting/pull/1427/files
SA PR: https://github.com/opensearch-project/security-analytics/pull/803/files
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by engechas — APPROVED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt [hunk 1]
```diff
@@ -13,19 +13,25 @@ class GetFindingsRequest : ActionRequest {
...
 
     constructor(
         findingId: String?,
         table: Table,
         monitorId: String? = null,
         findingIndexName: String? = null,
-        monitorIds: List<String>? = null
+        monitorIds: List<String>? = null,
+        severity: String? = null,
+        detectionType: String? = null
```
Comment (engechas) [inline src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt:26]: Should these be enums? Not sure if there's a fixed list of possibilities or not

File: src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt [hunk 2]
```diff
@@ -6,26 +6,42 @@ import org.opensearch.commons.alerting.model.Table
...
+import java.time.Instant
 
 class GetFindingsRequest : ActionRequest {
     val findingId: String?
     val table: Table
     val monitorId: String?
     val monitorIds: List<String>?
     val findingIndex: String?
+    val severity: String?
+    val detectionType: String?
```
Comment (engechas) [inline src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt:18]: Same comment on if these should be enums

**Review by engechas — APPROVED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt (hunk 2 above)
Comment (riysaxen-amzn) [reply src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt:18]: yes, that's right `detectorType` and `logType` are synonyms. We need to make it uniform across the codebase. There should be a seperate PR for that. Yes, severity is stored as `String`

Comment (riysaxen-amzn) [reply src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt:18]: Also, detectionType is is `different` from `detectorType/logType.` The detection type can threat-intel and rule based. The UI has a drop down for that and with this enhancement we are doing the filtering at the backend

File: src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt [hunk 3]
```diff
@@ -6,26 +6,42 @@ import org.opensearch.commons.alerting.model.Table
 import org.opensearch.core.common.io.stream.StreamInput
 import org.opensearch.core.common.io.stream.StreamOutput
 import java.io.IOException
+import java.time.Instant
 
 class GetFindingsRequest : ActionRequest {
     val findingId: String?
     val table: Table
     val monitorId: String?
     val monitorIds: List<String>?
```
[the remark this answers is not in the document]
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt:15]: `monitorId`, and `monitorIds`; but it's more of a nitpick; not a blocking comment.

File: src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt (hunk 2 above)
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/action/GetFindingsRequest.kt:18]: Ah thank you for clearing that up.

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**